cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java

```diff
@@ -375,18 +376,18 @@ String getCurrentVersion() {
   // 1. Either the application is not deployed before.
   // 2. If application is deployed before then the app artifact of the deployed application is not the latest one
   //    available.
-  private boolean shouldDeployApp(ApplicationId applicationId, SystemApplication application) throws Exception {
+  private boolean shouldDeployApp(ApplicationReference appRef, SystemApplication application) throws Exception {
```

Comment:
The SystemApplication and SystemProgram classes should have their version fields removed, and everything in these capability classes should work with ApplicationReference and ProgramReference.

They are not allowed to create applications of a specific version, so it doesn't make sense to have version in there.

Reply:
versions removed. Interesting thing is, CapabilityManagementServiceTest is deploying versioned system apps and programs... they're probably wrong